### ISSUE TITLE
Cannot connect to Dodo network with Darkstar 'Clearinghouse is down' at login screen

### ISSUE THREAD

**Issue by MidnightThunder** [devhawala/dodo#7]
I have setup Darkstar 1.1.9.0 with Viewpoint 2.06 NetCom and have configured Dodo which I am running with the default 
run-dodoserver.cmd and when I try to login I just get a message about Clearinghouse being down and the Star asks if I want to create a temporary local desktop. I have checked my configuration files multiple times and I can't see anything wrong with them. Darkstar is configured to connect to a Dodo Nethub on localhost with port 3333. I have tried turning of my Firewall and that did not help. I have attached my entire Dodo folder in the hope that someone more knowledgeable than me can help me fix the problem.

[Xerox Star Network.zip](https://github.com/devhawala/dodo/files/8496189/Xerox.Star.Network.zip)
![Xerox Star Network Config](https://user-images.githubusercontent.com/81908864/163577915-ce07563b-2b80-4ee4-8fe3-422bcb060378.PNG)

**Comment by devhawala** [comment]
Hello,

i did download your Dodo folder. All seems OK with your setup in the "dist" subfolder.

So i started the Dodo services on my Windows-10 test machine, first the nethub (run-nethub.cmd) and the server (run-dodoserver.cmd).
Then i started my Darkstar+VP-2.0.6 machine (this is Darkstar 1.1.9.0 with the VP2-disk from the distribution). After the long boot time (not so long as a real 8010 but much slower than VP 2.0.5 in my Draco emulation), i logged on with your account taken from the Clearinghouse configuration (there is only one user "mat") and it worked: the message was "Cannot find your Desktop on the file service. Select START in the option sheet to create a new one", confirming this as requested created a blank desktop.

I was able to send and receive a simple mail message (mail to "mat", no attachment) with this desktop, likewise i was able to store a VP document on the file service.

What failed was to copy the file back from the file service to the desktop: after sending a few bulk-data packets, the Dodo server started to resend packets, entering a strange endless resend loop (darkstar: "expecting #6 as next packet" -> dodo sends packets #6 and #7 -> darkstar: "expecting #6 as next packet" -> dodo sends packets #6 and #7 ...). I must investigate this when i find some free time, but this seems very strange!

Just to re-check, i did the same operations with my Darkstar+StarOS 5.0 machine, which had no problems at all (copying files back from the file service to the desktop worked fine), so the older StarOS works better than the newer VP2.0 ... the strangeness deepens!

As your error message with VP2.0.6 is the "clearinghouse not reached" flavour: does the Dodo log output show any activity when login on in VP?
- If yes: please post the Dodo log output up to the point where the message appears in Viewpoint.
- if no: please check if VP runs with the machine id used when you enabled all software options; maybe all software options are lost (new machine id and the grace period expired, whatever timeframe this is), so Viewpoint will not even allow to access the network, which would explain the "clearinghouse down" message...

Hope that helps, greetings, Hans

PS: on my Windows-10 box (about 8 years old hardware), the Darkstars run at ~120% speed when booting and ~170% when idle (no disk activity), according to the status line; if your machine is slower, increasing the resendDelay and sendingTimeGap values may help

PPS: if you intend to use the same Dodo installation also for Dwarf-based emulations, it would be a good idea to set the tuning values in the machines.cfg specifically for each machine; setting them in the dodo.properties slows down all accesses to the Dodo services, including emulations like Dwarf which don't need the slower network traffic.

**Comment by MidnightThunder** [comment]
By running "run-nethub.cmd" and "run-dodoserver.cmd" I have been able to login to the star and I am no longer getting the message "Clearinghouse is down".
However in the "directory" there is no Network divider which is strange because that should exist on a networked workstation (and I have selected the software options for VP NetCom). I think something very strange is occurring. As a side note the Star also never asked me to create a workstation administration desktop. I don't know if that's normal for a VP NetCom install or not. I did read the Viewpoint 2.0 software installation manual from bitsavers and am currently following the Quickstart training manual to setup my desktop.

**Comment by devhawala** [comment]
Hello,

the "Directory" icon should open with 3 entries, the last being "Network". If this is missing, this indicates that the Netcom software is not started. You could check the available software options: before login on, choose the menu option "List software options" on the top right menu (with the 3 bars) of the "Logon Option Sheet". Under "PRODUCT: Viewpoint", i have 4 active options:

![image](https://user-images.githubusercontent.com/34173994/163848436-21b19721-1486-4588-83d0-c9a1b45ef4ee.png)

If one of "Viewpoint" or "VP NetCom" is missing, your Viewpoint installation will be almost unusable, as some "invisible" software (not displayed in the "Loader") will not start and prevent other sotware depending on them from starting.
If so, then you should re-enable  the software options with the key found on the DarkStar readme page. (I don't known what the others are for, possibly needed too...)

As far as i remember, the workstation administration desktop is not available and necessary for networked VP workstations, as the user administration is expected to happen in the Clearinghouse and not locally as for stand-alone machines.

Greetings,
Hans

**Comment by MidnightThunder** [comment]
I have checked the software options and both "Xerox ViewPoint" and "VP NetCom" are listed. I checked again but I still don't have a Network divider in the "directory". Regarding the administration desktop, it's good to know that it is working as intended.

**Comment by devhawala** [comment]
Hello,

i have done some experiments, but was not able to make the "Network" entry disappear from the "Directory" on any of my  VP-2.0.x emulated machines (Draco or Darkstar).

As the network software is of a hidden type, it is not listed in the "Loader" (closed icon) resp. "Application Loader" (window title). So if the network software is not started for some reason, you can't see that it is not started or the reason why.

May be the only possible hint if it is running (except that the "Network" entry is missing) is to check if software that depends on it does run resp. can be started. I suppose that the "Mail" and "Remote Printing" software needs the network to run.

Are these 2 items in the "Loader" started on your machine?

Just asking, not knowing how to help...
Hans

**Comment by MidnightThunder** [comment]
I checked the "Loader" and both "Mail" and "Remote Printing" are running.

**Comment by MidnightThunder** [comment]
(Just came back to this after wanting to play with the Star again) 
I can't think of anything to fix this either, it is strange that it can connect to the network to login but then I can't see the network divider on the star. I wonder if it has something to do with me still using winpcap on windows 10, maybe I should try swapping it out for npcap, what do you think?

**Comment by devhawala** [comment]
Hello,

still no idea what may go wrong with your VP installation. But i don't think that switching the pcap implementation would change something, as the VP login already works.

BTW: does your login really connect to the Dodo server? Do log lines appear in the Dodo server output telling that your login was successful on the Dodo side?

**Comment by MidnightThunder** [comment]
Yes I think, I get messages in the Dodo server output suggesting it was a successful login.

Also on another note, is there a way to hard code the time the Dodo server will report to the Star so it can be automatically kept in the range for successful product factoring?
